### DISCUSSION TITLE
Contributing Developer: Problems with unit tests on local machine

### DISCUSSION THREAD

**Discussion by jose-ws** [renovatebot/renovate#9938] [Request Help]
Hello, 

I have followed the guide to get setup on my local machine and start contributing to renovate. I have installed the prerequisites and I have ran yarn to install and build.

The versions I currently have are:
git - 2.30.1
node- 15.11.0
yarn - 1.22.5
python - 3.8.9
java - 1.8.0_291

However when I run the tests on the fresh main branch with no changes I get several failed tests. The tests that fail for me are these: 

lib/versioning/regex/index.spec.ts (12.191 s)
  regex › throws › ^(?<major>\d+)?(?<!y)x$
  regex › throws › ^(?<major>\d+)?(?<=y)x$

lib/util/git/index.spec.ts (361.257 s)
  util/git/index › getFileList() › should exclude submodules
  util/git/index › isBranchStale() › should return false if same SHA as master
  util/git/index › getBranchCommit(branchName) › should return same value for equal refs
  util/git/index › getBranchParentSha(branchName) › should return sha if found
  util/git/index › mergeBranch(branchName) › should perform a branch merge
  util/git/index › getBranchLastCommitTime › should return a Date
  util/git/index › commitFiles({branchName, files, message}) › does not push when no diff
  util/git/index › getCommitMessages() › returns commit messages (snapshot)
  util/git/index › initRepo()) › should fetch latest (snapshot)

Any assistance would be appreciated.

**Comment by yoswein** [comment]
I have the same problem for the same tests.
The versions I use are:
git: 2.31.1.windows.1 (also reporoduced with version 2.27.0.windows.1)
node: v14.16.1
yarn: 1.22.10
python: 3.9.5
java: openjdk 11.0.9.1 2020-11-04


![image](https://user-images.githubusercontent.com/5477868/117684019-99417200-b182-11eb-9177-07769699c133.png)

**Comment by viceice** [comment]
I think the windows defender is interferring the git operations, i had that too some time ago. I also get errors when allowing jest to run on all virtual cores, so need to reduce them by half (`-w 8` )

**Reply by viceice** [reply]
That why im mostly use wsl2 for local development 😉

**Reply by rarkins** [reply]
https://docs.microsoft.com/en-us/windows/android/defender-settings

**Reply by yoswein** [reply]
I tried running on wsl2 and still have those errors
![image](https://user-images.githubusercontent.com/5477868/117829153-7d9ca100-b240-11eb-842a-6f5e0f8ddab0.png)


The link for the defender settings was mostly about excluding gradle and android caching which I doubt is my problem. I couldn't even test that since I Sophos and I couldn't find a way to exclude anything. 
Are those the only settings in the windows defender you think causing these problems?

**Reply by viceice** [reply]
maybe it's your sophos,  which is blocking something, we cantch change anything for it.

Please show the failing tests, your latst screenshot doesn't show any failed tests output, so we can't help solving them.

**Reply by rarkins** [reply]
Re: the link, I thought you could add your renovate source directory to the excluded folders, using the article as an example.

**Comment by yoswein** [comment]
The exclusion rule for Sophos didn't work either. 
I'm attaching the test log output. 
[tests.txt](https://github.com/renovatebot/renovate/files/6466972/tests.txt)

**Reply by rarkins** [reply]
One error is `unknown option `show-current'`. `show-current` was added to git 2.22.0 so it sounds quite likely that an older version of git is being used. Can you check that first?

```
  ● util/git/index › initRepo()) › should use extra clone configuration

    error: unknown option `show-current'
```

**Reply by yoswein** [reply]
@rarkins it worked! For some reason the git version in my wsl2 machine was lower than in the host machine. That being said, in the docs, there is no minimal git version like there is for Node, Yarn etc...

It still doesn't work well on my windows machine but at least it's working somewhere. 

Thanks a lot!

**Reply by rarkins** [reply]
If anyone has the time then it would be good to circle back to the non-WSL approach to see which tests fail and if they fail every time

**Comment by viceice** [comment]
It would be interesting which windows version are failing, I've 20H2 Wich is working fine.

Also the CPU and memory maybe relevant.

I've an AMD ryzen with 16 virtual cores and 32gb ram.

Maybe the memory is a big issue. I've also seen that limit jest worker to physical CPU core can solve failing git tests some time ago. (`-w 8` for me)
